subprojects/launcher/src/integTest/groovy/org/gradle/launcher/continuous/SmokeContinuousIntegrationTest.groovy

```diff
@@ -165,8 +175,14 @@ class SmokeContinuousIntegrationTest extends AbstractContinuousIntegrationTest {
         when:
         sources.file("sub/other/included.txt").createFile()
         then:
-        buildTriggeredAndSucceeded()
-        outputContains("includedFiles: 3")
+        if (OperatingSystem.current().linux) {
+            // On Linux, we don't watch sub, since it is a filtered directory,
+            // and it doesn't contain any file or unfiltered directory
+            noBuildTriggered()
+        } else {
+            buildTriggeredAndSucceeded()
+            outputContains("includedFiles: 3")
+        }
```

Comment:
This `if` is a bit misleading. The title of of the test is `detects changes in filtered file tree inputs`, but we are actually checking the opposite on Linux. I wonder if we should instead break this use case out into a separate test and add two separate test cases instead, one for `@Requires(Linux)` and another for `!Linux`. WDYT?

Reply:
I did split the test, which also shows that this part also doesn't work for hierarchical watchers.